libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/arrival/api/MapboxBuildingHighlightApi.kt

```diff
@@ -82,11 +114,50 @@ class MapboxBuildingHighlightApi(
                     .fillExtrusionHeight(heightExpression())
             )
         } else {
-            (style.getLayer(layerId) as FillExtrusionLayer)
+            style.getLayerAs<FillExtrusionLayer>(layerId)
                 .filter(selectedBuilding)
         }
     }
 
+    private fun updateFillBuildingsLayer(style: Style) {
+        val layerId = EXTRUDE_BUILDING_LAYER_ID
+        if (!extrudeBuildings) {
+            style.removeStyleLayer(layerId)
+            return
+        }
+        val buildingFillColorExpression = buildingFillColor(style) ?: return
+
+        val ids = highlightedBuildings.mapNotNull { it.feature.id()?.toLong() }
+        val notSelectedBuildings = not(inExpression(id(), literal(ids)))
```

Comment:
This is the relevant part of the logic to pull this off, right?

I'm wondering whether we should just expose this in some form, and for the rest of it let developers use the Maps SDK APIs directly so that we avoid wrapping too much of the Maps SDK surface. We can link to the right references/docs.

Highlighting the destination building is much more in the navigation domain than highlighting all buildings. I'd just like to avoid developers being confused about which API they should use or come to Nav SDK with feature requests to customize all the extruded buildings.

Reply:
Agree that extruding all buildings falls into the maps domain, and exposing the queried building could be preferred. This has also been requested for analytics purposes.

Some options considered: 
1. we could expose the `List<QueriedFeatures>`
2. or expose this an actual building object ([in maps or in navigation](https://github.com/mapbox/mapbox-maps-android/issues/235#issuecomment-841320263))

And then we can expose it through the highlightBuilding. Or we could create a `MapboxBuildingHighlightObserver`. This pull request is exposing it this way:
``` kotlin
MapboxBuildingHighlightApi 
    /**
     * Access the currently highlighted buildings.
     */
    var highlightedBuildings: List<QueriedFeature> = emptyList()
        private set

    fun highlightBuilding(
        point: Point?,
        callback: (List<QueriedFeature>) -> Unit = { }
...
```

Reply:
Addressed this. Moved the map extruding code to the examples.